src/react_native/utils.cljs

```diff
@@ -0,0 +1,20 @@
+(ns react-native.utils)
+
+(defonce ^:private throttle (atom {}))
+
+(defn- throttled-on-press
+  [{:keys [on-press allow-multiple-presses? throttle-duration]}]
+  (if allow-multiple-presses?
+    on-press
+    (fn []
+      (let [pressable-id (rand 10000)]
```

Comment:
@flexsurfer, @ilmotta this approach is not working everywhere. Some screens are causing component re-render and changing this pressable-id and allowing multiple button presses.
I will look more into this on how we can store state specific to that button etc. Means how we can track even if component is re-rendered we already have state for that button.
We don't have any id or something for these buttons. Props hash is not that useful, sometime same props or have anonymous functions etc in prop which changes on re-render.

Any ideas on how we can store state in this scenario?

Reply:
It's unclear to me what to do to solve this once and for all in non-intrusive/automatic way that performs well.

One idea would be to allow the function to receive the `pressable-id` instead of generating it randomly. That way, in problematic places that re-mount the button, the caller can pass a stable ID that will survive re-mounts. If we don't have that many exceptions in the repo, this can work very well both for maintainability and functionality. Clojure's namespaced keys could help us if we go this route.

```clojure
[button {:on-press    #(...)
         :throttle-id ::open-foo}]
```

But if we want to go the route of automatic IDs it's very trick, because we would need to generate something that uniquely identifies a button based on its values, but this would be too expensive. As you said, hashing is dangerously slow. Even hashing a small string a few times seems expensive in a quick check in CLJS.

Reply:
Found an interesting solution that automatically creates a unique ID per component using a hash of ref.
hash is comparatively slower then rand-int, So I used hash in setTimeout.
Below are the times taken by rand-int, hash, merge, and assoc.
```
      (print "--------------")
      (time (rand-int 100000))
      (time (hash ref-value))
      (time (merge {} {:a :b}))
      (time (assoc {} :a :b))
      (print "++++++++++++++")
```

<img src="https://github.com/status-im/status-mobile/assets/17097240/0209927a-a09f-4c53-80a3-2d7c62383a38" alt="Screenshot" height="300"/>

Reply:
Very interesting approach. I'll have another look @Parveshdhull

Reply:
BTW, your PR looks good with the new solution you found!